core/domain/draft_upgrade_services.py

```diff
@@ -312,6 +312,114 @@ def _convert_html_in_draft_change_list(
                 })
         return draft_change_list
 
+    @classmethod
+    def _convert_states_v52_dict_to_v53_dict(
+        cls, draft_change_list: List[exp_domain.ExplorationChange]
+    ) -> List[exp_domain.ExplorationChange]:
+        """Converts from version 52 to 53. Version 53 fixes general
+        state, interaction and rte data. This will update the drafts
+        for state and RTE part but won't be able to do for interaction.
+        The `ExplorationChange` object for interaction is divided into
+        further properties and we won't be able to collect enough info
+        to update the draft.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+
+        Raises:
+            InvalidDraftConversionException. The conversion cannot be
+                completed.
+        """
+        for exp_change in draft_change_list:
+            if exp_change.cmd != exp_domain.CMD_EDIT_STATE_PROPERTY:
+                continue
+            if (
+                exp_change.property_name ==
+                exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS
+            ):
+                # Here we use cast because this 'if' condition forces
+                # change to have type EditExpStateProperty
+                # InteractionAnswerGroupsCmd.
+                edit_interaction_answer_groups_cmd = cast(
+                    exp_domain.
+                    EditExpStatePropertyInteractionAnswerGroupsCmd,
+                    exp_change
+                )
+                answer_group_dicts = (
+                    edit_interaction_answer_groups_cmd.new_value
+                )
+                for answer_group in answer_group_dicts:
+                    answer_group['tagged_skill_misconception_id'] = None
+                    answer_group['outcome']['refresher_exploration_id'] = None
+
+                    if len(answer_group['rule_specs']) == 0:
+                        answer_group_dicts.remove(answer_group)
+                    if (
+                        answer_group['outcome']['labelled_as_correct'] and
+                        answer_group['outcome']['dest'] == exp_change.state_name
+                    ):
+                        answer_group['outcome']['labelled_as_correct'] = False
+            elif exp_change.property_name == (
+                exp_domain.STATE_PROPERTY_CONTENT
+            ):
```

Comment:
This can fit one line (without the brackets)

Reply:
Done